### PR TITLE
lvdocview: fix `getDocFormatName` implementation

### PR DESCRIPTION
**Pull request by benoit-pierre**
This change is [Reviewable](https://reviewable.io/reviews/koreader/crengine/609)

### CODE REVIEW

**Comment by poire-z** [comment]
What is it fixing ? Hard to see, looks like just reordering (why ? to keep them ordered by numeric value ?)

File: crengine/src/lvdocview.cpp [hunk 1]
```diff
@@ -4849,23 +4852,26 @@ const lChar32 * getDocFormatName(doc_format_t fmt) {
...
 	case doc_format_odt:
 		return U"OpenDocument (ODT)";
 	case doc_format_svg:
 		return U"SVG";
-	default:
-		return U"Unknown format";
+	case doc_format_md:
+		return U"Markdown";
 	}
+	return U"Unknown format";
```
Comment (Frenzie) [inline crengine/src/lvdocview.cpp:4874]: Is that preferred over default?

**Comment by Frenzie** [comment]
I think the assert added is the fix, though I'm not quite sure for what.

**Comment by benoit-pierre** [comment]
PDB and Markdown were not handled.

The assert ensure the value is part of the enum.

The no default case with the `doc_format_none` one ensure the compiler warn us if we add a new value to the enum and forget to update this function.

**Comment by benoit-pierre** [comment]
Reorder to match the enum order so scanning both visually is easier.

**Comment by Frenzie** [comment]
> The no default case with the doc_format_none one ensure the compiler warn us if we add a new value to the enum and forget to update this function.

It does? Hm, interesting.

**Review by poire-z — APPROVED** [review]

**Review by NiLuJe — APPROVED** [review]

**Comment by poire-z** [comment]
Bump as you feel like it with your base markdown PR (and bring my little fixes with it).

**Pull request merged**